google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -696,6 +696,7 @@ public String getOptimizerStatisticsPackage() {
     private CloseableExecutorProvider asyncExecutorProvider;
     private String compressorName;
     private String emulatorHost = System.getenv("SPANNER_EMULATOR_HOST");
+    private boolean userSpecifiedNumChannels = false;
```

Comment:
Do we need this variable? Can we directly check if `this.numChannels` is populated or not?

Reply:
There is no need of an extra variable. Used `this.numChannels` to validate if it is populated or not.